src/order/zorn.lean

```diff
@@ -386,6 +392,33 @@ theorem zorn_superset_nonempty {α : Type u} (S : set (set α))
 @zorn_nonempty_partial_order₀ (order_dual (set α)) _ S (λ c cS hc y yc, H _ cS
   hc.symm ⟨y, yc⟩) _ hx
 
+/-- Every chain is contained in a maximal chain. This generalizes Hausdorff's maximality principle.
+-/
+theorem chain.max_chain_of_chain {α r} {c : set α} (hc : zorn.chain r c) :
+  ∃ M, @zorn.is_max_chain _ r M ∧ c ⊆ M :=
+begin
+  obtain ⟨M, ⟨_, hM₀⟩, hM₁, hM₂⟩ :=
+    zorn.zorn_subset_nonempty {s | c ⊆ s ∧ zorn.chain r s} _ c ⟨rfl.subset, hc⟩,
+  { refine ⟨M, ⟨hM₀, _⟩, hM₁⟩,
+    rintros ⟨d, hd, hdM₀, hdM₁⟩,
+    have := hM₂ _ _ hdM₀,
+    induction this,
+    { exact hdM₁ hdM₀ },
+    change c ⊆ M with c ≤ M at hM₁,
+    exact ⟨le_trans hM₁ hdM₀, hd⟩ },
```

Comment:
Doing induction on equality is definitely something I'm not seeing often!
```suggestion
    rintro ⟨d, hd, hMd, hdM⟩,
    exact hdM (hM₂ _ ⟨hM₁.trans hMd, hd⟩ hMd).le },
```

Reply:
Haha, not sure how I missed that